### PR TITLE
[onert] Introduce APIs for asynchronous exeuction

### PR DESCRIPTION
**Pull request by wateret**
- Introduce `nnfw_run_async`
- Introduce `nnfw_await`

ONE-DCO-1.0-Signed-off-by: Hanjoung Lee <hanjoung.lee@samsung.com>

### CODE REVIEW

**Comment by wateret** [comment]
```
             +--------------+
             | INITIALIZED  |
             +--------------+
               |
               | load_model
               v
             +--------------+
             | MODEL_LOADED |
             +--------------+
               |
               | prepare
               v
             +--------------+
             |   PREPARED   | -+
             +--------------+  |
               |               |
               | run           |
               v               |
             +--------------+  |
  +--------> | FINISHED_RUN |  |
  |          +--------------+  |
  |            |               |
  | run_wait   | run_async     | run_async
  |            v               |
  |          +--------------+  |
  +--------- |   RUNNING    | <+
             +--------------+
```

File: runtime/onert/api/include/nnfw.h [hunk 1]
```diff
@@ -257,6 +257,37 @@ NNFW_STATUS nnfw_prepare(nnfw_session *session);
...
+ * session is prepared for inference by {@link nnfw_prepare}, set input and output buffers
+ * by {@link nnfw_set_input} and {@link nnfw_set_output}.</p>
+ *
+ * <p>This function returns immediately after starting a thread to run the inference.
+ * To get the result of it {@nnfw_run_wait} must be called</p>
+ *
+ * @param[in] session The session to run inference
+ * @return    @c NNFW_STATUS_NO_ERROR if successful
+ */
+NNFW_STATUS nnfw_run_async(nnfw_session *session);
```
Comment (hseok-oh) [inline runtime/onert/api/include/nnfw.h:273]: @wateret Can one session have many inference threads?

Comment (wateret) [reply runtime/onert/api/include/nnfw.h:273]: No. I chose not to support that for now.

Discussed from https://github.com/Samsung/ONE/issues/2541#issuecomment-648522313

> Let's support single asynchronous first, assuming that only one session is used asynchronously but sequentially.

Comment (hseok-oh) [reply runtime/onert/api/include/nnfw.h:273]: I see. Thanks.

Comment (wateret) [reply runtime/onert/api/include/nnfw.h:273]: @hseok-oh I would like keep discussing on https://github.com/Samsung/ONE/issues/2541#issuecomment-657505760 PTAL 😄

File: runtime/onert/api/include/nnfw.h [hunk 2]
```diff
@@ -257,6 +257,37 @@ NNFW_STATUS nnfw_prepare(nnfw_session *session);
...
+ *
+ * <p>When this function returns, it means that this session has finished
+ * the asynchronous run. Then the user can safely use the output data.</p>
+ *
+ * <p>This function return after the asynchronous inference is finished.</p>
+ *
+ * @param[in] session The session to run inference
+ * @return    @c NNFW_STATUS_NO_ERROR if successful
+ */
+NNFW_STATUS nnfw_run_await(nnfw_session *session);
```
Comment (hyunsik-yoon) [inline runtime/onert/api/include/nnfw.h:289]: shout the name be `nnfw_await`? `nnfw_run_await` has two verbs in name.

Comment (wateret) [reply runtime/onert/api/include/nnfw.h:289]: Hmm... that sounds better I guess

**Pull request closed without merging**